appeals/api/src/server/endpoints/documents/documents.controller.js

```diff
@@ -316,17 +316,15 @@ const updateDocumentsAvCheckStatus = async (req, res) => {
 	const { body } = req;
 	try {
 		const documents = body.documents;
+		await documentRepository.createDocumentAvStatus(documents);
+
 		for (const document of documents) {
 			const latestDocument = await documentRepository.getDocumentWithAllVersionsById(document.id);
-			if (!latestDocument) {
-				return res.status(404).send({ errors: { body: ERROR_NOT_FOUND } });
-			}
-			const versionList = latestDocument.versions?.map((v) => v.version);
-			if (!versionList || versionList.indexOf(document.version) < 0) {
-				return res.status(404).send({ errors: { body: ERROR_NOT_FOUND } });
+			const versionList = latestDocument?.versions?.map((v) => v.version);
+			if (versionList && versionList.indexOf(document.version) < 0) {
+				await documentRepository.updateDocumentAvStatus(document);
 			}
 		}
-		await documentRepository.updateDocumentAvStatus(documents);
```

Comment:
I can't work out if this should be in a transaction or not. I think as we create the AV status first, this might be OK.

Reply:
Yeah we write the AV result in the disconnected table first (line 319), then we want to update the DocumentVersion rows that we can find (to cover records that may already be in the DB), on line 325

Reply:
I think because `addDocument` uses a transaction when checking for scan status, this is OK.